airflow/models/taskinstance.py

```diff
@@ -204,6 +204,7 @@ def clear_task_instances(
                 ti.max_tries = max(ti.max_tries, ti.prev_attempted_tries)
             ti.state = State.NONE
             ti.external_executor_id = None
+            ti.clear_next_method_args()
```

Comment:
Should we add some sort of unit test for it?

suggestions @uranusjr @dstandish ?

Reply:
Some additional asserts in `TestClearTasks` should be good enough.